pkg/controller/pod.go

```diff
@@ -115,17 +109,10 @@ func (r *ReconcilePod) createReplicas(ctx context.Context, podKey string) error
 					return nil
 				}
 
-				numCreated := 0
-				// loop over all eligible nodes to create replicas.
-				for _, node := range nodes {
-					if numCreated >= azVolume.Spec.MaxMountReplicaCount {
-						break
-					}
-					if err := createReplicaAzVolumeAttachment(ctx, r, azVolume.Status.Detail.ResponseObject.VolumeID, node, azVolume.Spec.Parameters); err != nil {
-						klog.Warningf("Error creating %d/%d replicas azvolumeattachment for pod %s and volume %s on node %s. Error: %v", azVolume.Spec.MaxMountReplicaCount, numCreated, podKey, volume, node, err)
-						return err
-					}
-					numCreated++
+				err = r.controllerSharedState.manageReplicas(ctx, azVolume.Spec.UnderlyingVolume, r, r.namespace)
```

Comment:
Looking at the e2e test failures, it seems the reason no replicas are create is because `manageReplicas` can't find the `AzVolume` instance:

```log
2022-01-10T12:51:20.4438991Z I0110 11:55:39.870754       1 pod.go:87] Pod azuredisk-1386/azuredisk-volume-tester-8kpq4 has 1 volumes. Volumes: [pvc-0f90b76a-b9bf-4982-b581-893f04ea13a5]
2022-01-10T12:51:20.4441451Z I0110 11:55:39.870764       1 pod.go:75] Successfully created replicas for pod azuredisk-volume-tester-8kpq4. Reconciliation succeeded.
2022-01-10T12:51:20.4443075Z I0110 11:55:39.870813       1 common.go:1312] Aborting replica management... volume (pvc-0f90b76a-b9bf-4982-b581-893f04ea13a5) does not exist
2022-01-10T12:51:20.4445507Z I0110 11:55:39.870823       1 common.go:299] operation queue for volume (pvc-0f90b76a-b9bf-4982-b581-893f04ea13a5) fully iterated and completed.
```

Here `r.namespace` is passed to `managedReplicas` but it is not initialized by this reconciler because pods are not specific to the Mandalay driver's namespace. You'll need to pass the namespace here. It's not currently available from the shared state but should be. You can get it from the `driver.objectNamespace` field in `newDriverV2` when initializing the shared state

Reply:
@kassarl I noticed that we were inconsistently using the `--driver-object-namespace` parameter which allows specifying the CRD namespace and a hardcoded constant, so I went and complete support for customizing the CRD namespace in PR #340. It includes adding the CRD namespace to the shared state.